WG_GO_CORE.md

```diff
@@ -0,0 +1,177 @@
+
+## Current Projects
+* go-ipfs
+* IPLD
+* gx
+* Data Exchange/Transfer - improve performance of data transfer
+* Data Management - improve ability of Go IPFS to manage large data sets
+* Provider Strategies - improve efficiency of DHT use
+* Commands Lib
+* API
+* Base32 CIDv1
+
+## Major 2019 goals to achieve
+1. Benchmarks for Go IPFS that measure IPFS behavior across data lifecycle for a broad set of inputs. 
+2. ‘Indefinite’ refactors are complete; code is maintainable and understandable
+3. Improve data transfer speed so that it is at least 0.8 times that of bittorrent/rsync, (whichever is faster) for a given set of peers and data
+4. Support adding/getting/pinning large volumes of data efficiently. Go-IPFS can add (and provide) 10TB of small files (< 1MB each) at > 0.5 times the speed of a simple dd] copy. 
+5. Make storing, looking up, and retrieving provider data in the DHT reliable and fast. 
+6. Complete, documented, performant, stable APIs for interacting with a running Go IPFS node.
+
+## Milestones
+
+### 📦 Package managers
+
+* `M (P0)`: The IPFS core team understands the needs of package management communities and engages their interest
+   * Meet people from various package management communities -- both distros and language tooling -- and learn about their needs
+   * Support interested package manager communities with DRIs
+      * F-droid
+      * Guix
+      * Snapshots.debian.org ?
+* `M (P0)`: We have understanding and documentation of what kind of scale (in terms of size at one time; and change in size over time) mirroring a package manager archive requires.
+   * Ex: document how long mirroring npm into IPFS takes
+   * Ex: VictorBjelkholm/arch-mirror also exists, let’s make sure we gather stats
+* `M (P1)`: Develop documentation for what Package Distribution with IPFS should look like.
+   * There is documentation and guidelines to help maintainers from existing package management communities learn about IPFS, explore common alternatives, and figure out what would work for their system
+   * Document and highlight best practices for dealing with the common hurdles in moving to immutable-first: e.g. “save packages in a merkle tree; build a package index merkle tree; use IPNS to handle updates of the tip of package index tree”.
+* `M (P1)`: There is a way to prioritize different peers to optimize network utilization
+   * Bitswap peer prioritization supports pinning local / responsive peers 
+
+
+### 🗂 Large Files
+
+Go-IPFS is fast, efficient and stable when handling and transferring large amounts of data. (“large files” = 10GB - 1TB, “many small files” = 10K-1M 1MB files). The goals in this section are exclusive of finding the data (DHT/providers etc.). This is just the speed of transferring data when already connected to IPFS nodes that have that data.
+
+* `M (P0)`: Awesome go-ipfs benchmark test suite exists comparing ipfs performance and transfer size relative to bittorrent, rsync, http, cp, dd

```

Comment:
Benchmarks are super important for improving performance (aka optimizing the slow points), so we should absolutely prioritize at least a first draft of this (against a representative one of these - maybe bittorrent?) in Q1.

```suggestion
* Q1 - `M (P0)`: Awesome go-ipfs benchmark test suite exists comparing ipfs performance and transfer size relative to bittorrent
* Q2 - `M (P0)`: Expand benchmark test suite to compare relative to rsync, http, cp, dd
```